ambry-network/src/test/java/com.github.ambry.network/SelectorTest.java

```diff
@@ -94,6 +92,20 @@ public void testClientDisconnect()
   }
 
   /**
+   * Validate that a closed connectionId is returned via disconnected list after close
+   */
+  @Test
+  public void testDisconnectedListOnClose()
+      throws Exception {
+    String connectionId = blockingConnect();
+    selector.close(connectionId);
+    selector.poll(0);
```

Comment:
perhaps also add a check to ensure that a call to disconnected() before the poll does not return the connection id?